xdsl/dialects/arm/assembly.py

```diff
@@ -9,3 +12,19 @@ class AssemblyInstructionArg(abc.ABC):
     @abc.abstractmethod
     def assembly_str(self) -> str:
         raise NotImplementedError()
+
+
+class reg(AssemblyInstructionArg):
+    """
+    A wrapper around SSAValue to be printed in assembly.
+    Only valid if the type of the value is a RegisterType.
+    """
+
+    value: SSAValue
+
+    def __init__(self, value: SSAValue) -> None:
+        self.value = value
+
+    def assembly_str(self) -> str:
+        assert isinstance(self.value.type, RegisterType)
+        return self.value.type.register_name.data
```

Comment:
why is this in the arm assembly file if it is supposed to be used with RegisterType and not ArmRegisterType?

Reply:
Two reasons, one is that I'd like to move this file to somewhere to be shared by all dialects, but also because the ArmRegisterType imports this file, and circular imports aren't allowed. We could have moved this class into that file, I don't feel super strongly about which is better.